### PR TITLE
feature-cmake-docs: restructure the build system pages

### PR DESCRIPTION
**Pull request by 0xc0170**
This is first in series to our feature-cmake-docs. We will work on the feature branch to prepare CMake documentation. To start with, we first need to address how the old build system is described.

It was part of Mbed OS, it is referenced in multiple places and uncovering lot of details that can lead to confusion as we are introducing different build system. We should consolidate these and put two different build systems under separate folders (docs pages).

This PR in detail:
- create build-system folder under mbed_cli_1 where we put all build system pages from the docs
- what stays is generic information (program setup or quick setup should not be about what build system dictates but what works for anyone). And only reference build system details (cli 1 or cli 2 or anything else).

 docs/tools/mbed_cli_1/build-system/build.md is the main landing page for python build tools.

I left few TODO_CMAKE, as we will restructure number of docs, we will find out later  what the link actually is. We will use these until we get the structure right and approved, and fix all refs. Note, I don't recall how the .html are actually rendered so need some help.

### CODE REVIEW

File: docs/debugging-testing/testing/testing_intro.md [hunk 1]
```diff
@@ -25,3 +25,55 @@ Greentea, `htrun` and `mbed-ls` are testing tools written in Python. Greentea te
 The testing system automates the process of flashing Mbed boards, driving the tests and accumulating test results into test reports. Developers and Mbed Partners use this system for local development, as well as for automation in a Continuous Integration environment.
 
 This section explains how to build, run and write tests. It also includes details about our Greentea and utest testing tools.
+
+## Test directories
+
+Functional tests are organized into test cases and test suite directories within a `TESTS` directory. Each test suite is a subdirectory of the `TESTS`, and each test case is a subdirectory of a test suite. When tests build, each test case compiles independently. The test suite `host_tests` is reserved for scripts that run and validate a test case. The following tree is a reduced version of the tests subdirectory of Mbed OS:
```
Comment (rajkan01) [inline docs/debugging-testing/testing/testing_intro.md]: Can we move the Unittest directory structure into `unit_testing.md` before this section https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/debugging-testing/testing/unit_testing.md#test-code-structure better
Even move greentea directory structure into `testing_greentea.md` before this section https://github.com/ARMmbed/mbed-os-5-docs/blob/development/docs/debugging-testing/testing/testing_greentea.md#test-code-structure

File: docs/tools/mbed_cli_1/build-system/build.md [hunk 2]
```diff
@@ -1,4 +1,77 @@
-# Build rules
+# Python build tools
```
Comment (rajkan01) [inline docs/tools/mbed_cli_1/build-system/build.md]: Can we name it "Python-based build system"?

Comment (0xc0170) [reply docs/tools/mbed_cli_1/build-system/build.md]: I'll rename

File: docs/debugging-testing/testing/testing_intro.md (hunk 1 above)
Comment (0xc0170) [reply docs/debugging-testing/testing/testing_intro.md]: I'll have a look now

Comment (0xc0170) [reply docs/debugging-testing/testing/testing_intro.md]: +1, fixed now

File: docs/tools/mbed_cli_1/build-system/build.md (hunk 2 above)
Comment (0xc0170) [reply docs/tools/mbed_cli_1/build-system/build.md]: fixed

**Review by rajkan01 — APPROVED** [review]
LGTM

**Comment by 0xc0170** [comment]
I'll merge this to the feature branch

**Pull request merged**